cache.js

```diff
@@ -0,0 +1,62 @@
+const generateDefKey = (data, to) => {
+  if (
+    !data ||
+    data.length === 0 ||
+    (!isArrayBuffer(data) && typeof data !== "string")
+  ) {
+    return;
+  }
+  let dataStr = data.toString("hex");
+
+  // Trim leading `0x` if present
+  if (dataStr.slice(0, 2) === "0x") {
+    dataStr = dataStr.slice(2);
+  }
+
+  return `${to}${dataStr.slice(0, 8)}`;
+};
+
+const getCachedDef = (tx, cache) => {
+  return cache.get(generateDefKey(tx.data, tx.to));
+};
+
+const saveDefToCache = (tx, cache, def, maxSize = 100) => {
+  if (!tx.data || tx.data.length === 0) {
+    return;
+  }
+
+  const key = generateDefKey(tx.data, tx.to);
+  if (!key) {
+    return;
+  }
+
+  if (cache.size >= maxSize) {
+    cache.delete(cache.keys().next().value);
+  }
+  cache.set(key, def);
```

Comment:
Just to be crystal clear here, please confirm this API does the following:

1. `cache.next()` returns the oldest item
2. `cache.set()` adds it as the newest item

Reply:
Correct.